File: API/src/infra/mongoose/repositories/CalculatorRepository.ts
import { Calculator } from "../../../domain/entities/Calculator";
import ICalculatorRepository from "../../../domain/interfaces/repositories/ICalculatorRepository";
import CalculatorMapper from "../mappers/CalculatorMapper";
import CalculatorModel from "../models/CalculatorModel";

export class CalculatorRepository implements ICalculatorRepository{
    
    async getById(id: string): Promise<Calculator | null> {
        const resultSchema = await CalculatorModel.findById(id);
        return CalculatorMapper.toEntity(resultSchema);
    }
    
    async save(calculator: Calculator): Promise<Calculator | null> {
        const schema = CalculatorMapper.toSchema(calculator);
        const resultSchema = await CalculatorModel.create(schema);
        return CalculatorMapper.toEntity(resultSchema);
    }
}

File: Worker/src/Worker.Infra.Data/Repositories/CalculatorRepository.cs
using MongoDB.Driver;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Worker.Domain.Entities;
using Worker.Domain.Interfaces;
using Worker.Infra.Data.Context;

namespace Worker.Infra.Data.Repositories
{
    public class CalculatorRepository : ICalculatorRepository
    {
        private readonly IMongoCollection<Calculator> DbSet;

        public CalculatorRepository(MongoContext context)
        {
            DbSet = context.GetCollection<Calculator>("calculator");
        }

        public async Task<Calculator> GetById(string id)
        {
            return await DbSet.Find(c => c.Id == id).FirstOrDefaultAsync();
        }

        public async Task Save(string id, Calculator calculator)
        {
            await DbSet.ReplaceOneAsync(c => c.Id == id, calculator);
        }
    }
}
